File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        n = len(nums)
        fps, curr = 0,0
        fp = [0]*n

        for i , x in enumerate(nums):
            if fp[i]:
                curr ^= 1
            
            if x^curr == 0:
                fps += 1
                curr ^= 1

                if i + k > n:
                    return -1
                if i + k < n:
                    fp[i+k] = 1
        return fps

File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.go
func minKBitFlips(nums []int, k int) int {
    n := len(nums)
    fc := 0
    fq := make([]int, 0)

    for i:= 0; i<n ; i++ {
        if len(fq) > 0 && i >= fq[0]+k {
            fq = fq[1:]
        }

        if (nums[i] + len(fq)) %2 == 0 {
            fc++
            fq = append(fq, i)
            if  i+k > n {
                return -1
            }
        }
    }
    return fc
}